### DISCUSSION TITLE
MOROS 0.11.0

### DISCUSSION THREAD

**Discussion by vinc** [vinc/moros#711] [General]
![moros](https://github.com/user-attachments/assets/d27637a7-5934-4d21-8362-99b0a275ccde)

[MOROS](http://moros.cc) is a text-based hobby operating system targeting computers with a x86-64 architecture and a BIOS. It draws inspiration from Unix and DOS to provide a simple and efficient environment for computing and communication from the command line.

## What's new in MOROS 0.11.0?

- Fixed binary loading 
- Fixed block bitmap area calculation
- Added a VGA graphics mode
- Added device files to configure the network stack
- Added device files to configure the VGA driver
- Added device files to configure the sound driver
- Added a package manager `pkg`
- Added an image viewer `render`
- Added a search mode to the text editor and the text viewer
- Added scrollback buffer support to the shell
- Added a `diff` command
- Added more syscalls and userspace programs
- Refactored the kernel into modules
- Replaced the `base64` command with `encode` and `decode` 
- Replaced the `fetch` command by `get`

Read the full list of changes on the [CHANGELOG](https://github.com/vinc/moros/blob/v0.11.0/CHANGELOG.md)

## How to run MOROS?

The binary can be booted from USB on a computer with a BIOS or from a virtual machine with Bochs, QEMU, or VirtualBox. Here's a quick way to do it with QEMU:

```
qemu-system-x86_64 -m 32 -cpu max -nic model=e1000 -soundhw pcspk -hda moros-v0.11.0.bin
```

Read more detailed instructions in the [README](https://github.com/vinc/moros/blob/v0.11.0/README.md)

<hr /><em>This discussion was created from the release <a href='https://github.com/vinc/moros/releases/tag/v0.11.0'>MOROS 0.11.0</a>.</em>

**Comment by viktorpopp** [comment]
How did you keep yourself motivated? Also what would you recommend doing after completing Philip Opperman's tutorial?

**Reply by vinc** [reply]
After 4 years on the project, I have to admit that I sometimes lose motivation, but the great thing about an OS is that there are so many different areas to explore. I can simply switch to another part of the kernel or the userspace and keep improving it!

After completing Phil's tutorial, my initial focus was on getting some core components working: a filesystem, a shell, and a text editor. Then I moved on to adding a networking stack and real hardware support. Also games, and a lisp!

Since then, I've worked on whatever features I needed to make MOROS more useful for my daily use, and the TODO list will probably never be empty! Using it every day has been a powerful motivator. I've come to see this as a slow, lifelong personal hobby project.
